packages/button/src/vaadin-lit-button.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { html, LitElement } from 'lit';
+import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { TooltipController } from '@vaadin/component-base/src/tooltip-controller.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { ButtonMixin } from './vaadin-button-mixin.js';
+import { buttonStyles } from './vaadin-button-styles.js';
+
+/**
+ * LitElement based version of `<vaadin-button>` web component.
```

Comment:
Not source code related, but wanted to create discussion without endless comments within the main PR about one topic. 

Something to consider from an outsiders perspective: As far as I understand the codebase and communication to the community, all usage of Lit element instead of polymer is a internal detail thanks to the removal of the public API, allowing to gradually do this within minor releases(?). Otherwise it could literally take years to get this on the road, looking at the public available comments that with the new and simplified release model of the platform the amount of major releases should be reduced. Vaadin 24 had to be pushed because of external factors like the Jakarta Namespace and Spring Boot / ecosystem end of life. I (hopefully) don't see any other big problem in the next 1 or even 2 years justifying another major release - which in turn would move this change (if it has to be done within a major release) to like 2024/2025.. leaving us/you with three options: 
- pushing at least some components to lit in 24 to get early Feedback like components as button, layouts and non-complex stuff and gradually move the others afterwards in minors to come (with the experience you get from Feedback of 24.0's components)
- doing all of this within minor versions of 24.x 
- Having this transformation wait for the next big release (25.0) with no definitive timeline 

With the official removal /  payed support of PolymerTemplate in flow for 24.0 - I think it would be the perfect timing to get this done / on the road.

Reply:
What knoobie said.

And also, what and where and who can I pay to increase velocity on this?

I've already talked to my team JS wizard @anoblet about contributing some of our time here, but we could find more budget.

Reply:
Thanks for your opinions! This is exactly the kind of discussion that I'd like to have.

>  all usage of Lit element instead of polymer is a internal detail thanks to the removal of the public API, allowing to gradually do this within minor releases(?). Otherwise it could literally take years to get this on the road

Yes, this is what the original plan was back in the day, described in the blog post from 2019 that I linked above.
I totally agree that from Java / Flow developer's point of view, these changes are purely internal refactorings.

However, we also need to consider Hilla and web components users who build their applications with Lit.
As now when we have React wrappers, not breaking them is also something that we have to consider.

Essentially, changing the library used under the hood is a behavior altering change, because of switching from sync DOM rendering / updates (Polymer) to async (Lit) - see my additions of `await nextFrame()` in unit tests.

Overall, I feel like this change is long overdue. We definitely shouldn't postpone it till 2024 or further.